README.md

```diff
@@ -15,14 +15,25 @@ The following example `marathon.json` fragment will read Vault path `/secret/abc
   },
   "secrets": {
     "secret_ref": {
-      "source": "/secret/abc/xyz@password"
+      "source": "/abc/xyz@password"
     }
   }
 }
 ```
 
 If the provided Vault path or field is not found, the environment variable will not be set. The same applies when it cannot be read because of permissions or other types of errors. Either way, it will be logged as an error in Marathon logs.
 
+The path in the secret source can be absolute or relative and it depends on format of the secret source. The path is absolute if the secret source starts with `/` otherwise is relative path.
+Both paths have a root defined in configuration (`sharedRoot` for absolute path and `appSpecificRoot` for relative path). 
+
+### Absolute path to a secret
+
+For the absolute path is a path to the vault defined as `<sharedRoot>/<path from the secret source>` (e.g. `secret/shared/database@password`). This kind of secret allows you to share secrets between applications.
+
+### Relative path to a secret
+
+For the relative path is a path to the vault defined as `<appSpecificRoot>/<marathon path and service name>/<path from the secret source>` (e.g. for application `test/myTestApp` it is `secret/private/test/myTestApp/database@password`). This concept will guarantee that secrets can not be read from other applications.

```

Comment:
The example how the path is constructed should include all the final path parts - the <appSpecificRoot> configuration, Marathon application path and the secret source.